lib/utlis.ts

```diff
@@ -0,0 +1,66 @@
+type MetadataProps = {
+  title?: string
+  description?: string
+  canonical: string
+  ogImage?: string
+}
+
+export const constructMetadata = ({
+  title,
+  description = "The best AI-powered code editor",
+  canonical = "/",
```

Comment:
Let not default the canonical to home. It should be whatever the page url is unless multiple urls point to the same page

Reply:
Sorry, can you explain this once. the canonical prop is mandatory. so, i think its fine to leave it like this